Sources/MapboxMaps/Annotations/Generated/CircleAnnotationManager.swift

```diff
@@ -206,36 +205,18 @@ public class CircleAnnotationManager: AnnotationManager {
         }
     }
 
-    // MARK: - Selection Handling -
+    // MARK: - Tap Handling -
 
     /// Set this delegate in order to be called back if a tap occurs on an annotation being managed by this manager.
-    public weak var delegate: AnnotationInteractionDelegate? {
-        didSet {
-            if delegate != nil {
-                setupTapRecognizer()
-            } else {
-                guard let view = view, let recognizer = tapGestureRecognizer else { return }
-                view.removeGestureRecognizer(recognizer)
-                tapGestureRecognizer = nil
-            }
-        }
-    }
+    public weak var delegate: AnnotationInteractionDelegate?
 
-    /// The `UITapGestureRecognizer` that's listening to touch events on the map for the annotations present in this manager
-    public var tapGestureRecognizer: UITapGestureRecognizer?
+    @objc internal func handleTap(_ tap: UITapGestureRecognizer) {
 
-    internal func setupTapRecognizer() {
-        let tapRecognizer = UITapGestureRecognizer(target: self, action: #selector(handleTap(_:)))
-        tapRecognizer.numberOfTapsRequired = 1
-        tapRecognizer.numberOfTouchesRequired = 1
-        view?.addGestureRecognizer(tapRecognizer)
-        tapGestureRecognizer = tapRecognizer
-    }
+        guard let delegate = delegate else { return }
```

Comment:
Since qrf is async, I wouldn't capture a strong ref to the delegate since that could result in messages to the delegate after it's been set to nil. The check here should just be an optimization to skip qrf if we know the delegate is nil.

Reply:
! , thank you.